src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -14,6 +14,7 @@ public static class ValidationUtilities
     {
         private static readonly string XmsPageable = "x-ms-pageable";
         private static readonly Regex TrackedResRegEx = new Regex(@".+/Resource$", RegexOptions.IgnoreCase);
+        private static readonly Regex resourceProviderPathPattern = new Regex(@"/providers/(?<resPath>[^{/]+)/", RegexOptions.IgnoreCase);
```

Comment:
If I understand correctly, the string following``` providers/``` is your resource provider, everything else should be resourcetype and resourcetypename. In other words, a URL should have one resource provider only, Am I missing something here?